src/sqlfluff/rules/L054.py

```diff
@@ -88,6 +91,11 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
         if not context.segment.is_type("groupby_clause", "orderby_clause"):
             return None
 
+        # Ignore Windowing clauses
+        for parent in context.parent_stack:
```

Comment:
I think you could simplify this a bit with the functional API (it's good at eliminating loops):
```
if context.functional.parent_stack.select(sp.is_type(*_self._ignore_types)):
    return LintResult()
```

Reply:
Done. Also changed the LintResult to return Memory like the others.